### DISCUSSION TITLE
ExceptionAdapter wrapping

### DISCUSSION THREAD

**Discussion by kalgon** [resteasy/resteasy#3080] [General]
Hi, I noticed that when using automatic type conversion for parameters (with a static method `valueOf(String)` or a constructor which takes a `String` as parameter), all exceptions are wrapped in an `ExceptionAdapter` (even runtime exceptions), this makes it difficult for those errors to be handled in an `ExceptionMapper` as they will be wrapped in an `InvocationTargetException` which will itself be wrapped in an `ExceptionAdapter`... Was there a good reason to wrap everything inside an `ExceptionAdapter` and not unwrapping `InvocationTargetException` in `TypeConverter`?

**Comment by jamezp** [comment]
What version of RESTEasy are you using?

**Reply by kalgon** [reply]
I am using 4.7.5

**Reply by kalgon** [reply]
But it's still the case on the [main branch](https://github.com/resteasy/resteasy/blob/main/resteasy-core/src/main/java/org/jboss/resteasy/util/TypeConverter.java#L272).

**Reply by jamezp** [reply]
Sorry for the long delay. I was on PTO and am catching up now.

I'm not sure why it was done like that, but it's been that way for at least 14 years :).  Where are you trying to use an `ExceptionMapper`? I know while processing exception mappers the exceptions are unwrapped.
